packages/jsts/tests/parsers/ast.test.ts

```diff
@@ -275,6 +275,66 @@ describe('ast', () => {
     const deserializedProtoMessage = deserializeProtobuf(serialized);
     compareASTs(protoMessage, deserializedProtoMessage);
   });
+
+  test('should support TSModuleDeclaration nodes', async () => {
+    const code = `namespace Foo { let a = 42;}`;
+    const ast = await parseSourceCode(code, parsersMap.typescript);
+    const protoMessage = visitNode(ast as TSESTree.Program);
+
+    expect(protoMessage.program.body[0].type).toEqual(
+      NODE_TYPE_ENUM.values['TSModuleDeclarationType'],
+    );
+    const tSModuleDeclaration = protoMessage.program.body[0].tSModuleDeclaration;
+    expect(tSModuleDeclaration.kind).toEqual('namespace');
+    expect(tSModuleDeclaration.id.type).toEqual(NODE_TYPE_ENUM.values['IdentifierType']);
+    expect(tSModuleDeclaration.id.identifier.name).toEqual('Foo');
+    expect(tSModuleDeclaration.body.type).toEqual(NODE_TYPE_ENUM.values['TSModuleBlockType']);
+    expect(tSModuleDeclaration.body.tSModuleBlock.body[0].type).toEqual(
+      NODE_TYPE_ENUM.values['VariableDeclarationType'],
+    );
```

Comment:
what is this `tSModuleDeclaration` ? is it a class or a plain object? If its an object and you are asserting many properties, would it be more pleasant to compare it like so? With this setup, it's more visible to the reader of the code of the exact nesting and/or structure. However, as I was typing this out, it did feel a bit longish..
```suggestion
    expect(protoMessage.program.body[0].tSModuleDeclaration).toStrictEqual({ // or toEqual or toMatchObject
      kind: 'namespace',
      id: {
        type: NODE_TYPE_ENUM.values['IdentifierType'],
        identifier: {
          name: 'Foo',
        }
      },
      body: {
        type: NODE_TYPE_ENUM.values['TSModuleBlockType'],
        tSModuleBlock: {
          body: [
            {
              type: NODE_TYPE_ENUM.values['VariableDeclarationType']
            }
          ]
        }
      }
    });
```